docs/reference/specifications/providers.md

```diff
@@ -0,0 +1,251 @@
+| socketPath         | FLAGD_SOCKET_PATH          | alternative to host port, unix socket                                           | String                   | null                          | rpc & in-process    |
+| certPath           | FLAGD_SERVER_CERT_PATH     | tls cert path                                                                   | String                   | null                          | rpc & in-process    |
+| deadline           | FLAGD_DEADLINE_MS          | deadline for unary calls, and timeout for initialization                        | int                      | 500                           | rpc & in-process    |
+| streamDeadlineMs   | FLAGD_STREAM_DEADLINE_MS   | deadline for streaming calls, useful as an application-layer keepalive          | int                      | 600000                        | rpc & in-process    |
+| retryBackoffMs     | FLAGD_RETRY_BACKOFF_MS     | initial backoff for stream retry                                                | int                      | 1000                          | rpc & in-process    |
+| retryBackoffMaxMs  | FLAGD_RETRY_BACKOFF_MAX_MS | maximum backoff for stream retry                                                | int                      | 120000                        | rpc & in-process    |
+| retryGraceAttempts | FLAGD_RETRY_GRACE_ATTEMPTS | amount of stream retry attempts before provider moves from STALE to ERROR state | int                      | 5                             | rpc & in-process    |
+| keepAliveTime      | FLAGD_KEEP_ALIVE_TIME_MS   | http 2 keepalive                                                                | long                     | 0                             | rpc & in-process    |
+| cache              | FLAGD_CACHE                | enable cache of static flags                                                    | String - lru, disabled   | lru                           | rpc                 |
+| maxCacheSize       | FLAGD_MAX_CACHE_SIZE       | max size of static flag cache                                                   | int                      | 1000                          | rpc                 |
+| selector           | FLAGD_SOURCE_SELECTOR      | selects a single sync source to retrieve flags from only that source            | string                   | null                          | in-process          |
+| contextEnricher    | -                          | sync-metadata to evaluation context mapping function                            | string                   | identity function             | in-process          |
+
+### Custom Name Resolution
+
+Some implementations support [gRPC custom name resolution](https://grpc.io/docs/guides/custom-name-resolution/), and abstractions to introduce additional resolvers.
```

Comment:
> Some implementations support...

In the provider spec we should clarify if this feature is optional. But if it's implemented it should be consistent across implementations.